src/mongo/editors/MongoFindOneResultEditor.ts

```diff
@@ -34,12 +34,15 @@ export class MongoFindOneResultEditor implements ICosmosEditor<IMongoDocument> {
 
     public async update(newDocument: IMongoDocument): Promise<IMongoDocument> {
         const node = await this._tree.findNode(this.id);
+        let result: IMongoDocument;
         if (node) {
-            return (<IAzureNode<MongoDocumentTreeItem>>node).treeItem.update(newDocument);
+            result = await (<IAzureNode<MongoDocumentTreeItem>>node).treeItem.update(newDocument);
```

Comment:
This is the other issue I saw.  Why was the return removed here?

Reply:
My bad. We need to kick off the refresh before returning, but after updating the value. 
Should return the result.